### DISCUSSION TITLE
[Feature] Add a new section with recently added photos

### DISCUSSION THREAD

**Discussion by imler** [immich-app/immich#8716] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, there is no way to see new photos with an older date that was recently added to the Immich.

I sometimes add some older photos that I gathered from other family members and, because all photos are arranged by dates, they land somewhere deep in the timeline and it’s hard to find by other users.

Some of them might even don’t know that there is something new if I didn’t tell them, because at first it’s buried and not visible.

Would be great if there were a separate page where all recently added photos would be visible by the time of adding.

I’ve read somewhere in other discussions that supposedly something like that was already present but was removed for some reason. I didn't use Immich back then, so I don't know if it's true and if so what was the reason.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by guipace** [comment]
+1

**Comment by nothing2obvi** [comment]
I would also really appreciate this feature, for the same reasons as @imler. It's very difficult to to perform actions (e.g. sort into albums) photos that were recently added but are now deep in the timeline.

**Comment by AFumi39** [comment]
Yes please, add this functionality... It's a noticeable flaw of a great software like Immich.
I used this feature a lot back in Google Photo and then in Synology Photo. I'm trying to switch to Immich, but this is a must-have 🙇‍♂️

**Comment by Acidiphilium** [comment]
I agree, that would be very useful to find older newly added photos.

**Reply by ckonkel** [reply]
As a supporter I agree

**Comment by not4smurf** [comment]
Yes please!

I take most of my photos with my phone and they upload automatically and everything is great.  But, I also have a "real camera" that I pick up from time to time if I see an interesting bird in the garden, a nice sunset or something.  I tend to forget about these photos and every month or so I'll pull the card from the camera and upload them - and they're lost!

My current work around is to filter my library by camera make and model and these will be the newest from that camera.

**Comment by maelstrm-code** [comment]
This would be exceptionally helpful, so that we can identify images uploaded in error and bulk delete them. Ideally the feature should allow us to select all photos uploaded in a date/time range. I've accidentally uploaded a large number of screenshots and other garbage due to an incorrect path in immich-go and it seems the only answer at the moment is to wipe Immich and start again which would be tragic!

**Comment by xuedi** [comment]
I am not a node developer, i imagine something like:
 - import button is pressed & lastImported.json gets flushed
 - after each import the function/worker, add the file (or new image) to lastImported.json
 - button in menu just send the static lastImported.json into the frontend magic (no idea how it works)

**Comment by rajeshmukkala** [comment]
+1

**Comment by coltenkrauter** [comment]
Note that in mobile, there is a search view provided that is supposed to show recently added. It doesn't seem to be working properly in my experience, so I made [this bug report](https://github.com/immich-app/immich/issues/17756).

**Reply by coltenkrauter** [reply]
Update.... [A contributor shared, ](https://github.com/immich-app/immich/issues/17756#issuecomment-2819115876)

> I can reproduce this. Pretty sure the solution would be to rename it to Recently Taken , which I believe was the intention of this. We don’t have any “by added date” feature/API endpoint available or planned as far as I know.

![image](https://github.com/user-attachments/assets/2db71cd6-aa73-4d04-aae3-0441c30a9d38)

**Comment by OnixGH** [comment]
Another use case: an immich album is shared via a web link via which a group of people can view/upload media (of recurring events). However, uploads come from many different people at different times, so only sorting by "taken" timestamp makes it impossible to see what was added and forces going through the album again (and occasionally photos from further back are uploaded, so those are then missed by most as well).

Being able to sort an album by upload date --- or perhaps hat the "other" app has, an "activity" timeline (that shows new uploads/comments) -- would solve this.

Additionally it would be nice if web users could choose/override the album's sort order locally (e.g. via a cookie) so they don't need album sorting administrative rights to switch sorting orders.

**Comment by eliasfloreteng** [comment]
I found that I can use the upload from computer option when adding to an album to "group" photos I upload at the same time

**Comment by jagfirerwalker** [comment]
Is there any update on this? I love that feature as well

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by mingqlin** [comment]
I am also waiting for this feature, it is very helpful, whenever I import old photos, it will be hard to find on timeline, with recently added photos, it will be much easier to find all newly imported photos no matter what dates the photo is taken

**Comment by Sulfiore** [comment]
Just had the need of this feature. Uploaded thousands of pictures, and i don't know which ones because there is no recently added view. Maybe an easy way to implement this would be to sort by added time maybe

**Comment by goldmund256** [comment]
Until this gets hopefully one day implemented, here is a workaround: create an album called IMPORT and upload all new images there.

`$ immich-go upload from-folder --into-album IMPORT --server https://example.com --api-key=asdf1234 *`

**Reply by AFumi39** [reply]
That's only works if you volunteering uploads the images knowing that... And you have to remove them from album after checking... Makes no sense to me and not my use case.
Than I prefer to assign all images with no album to a "Dummy" album, and than I only need to check for new images without album. Then assign to albums or to Dummy when checked.
But that's a workaround, not a solution

**Reply by goldmund256** [reply]
Yep, it's a **workaround**. I have edited my comment.

Since I am the only person ever uploading to Immich, it works for me. Of course, a proper section with recently added photos would be absolutely fantastic.

**Comment by sblock23** [comment]
Please add this feature. It would be useful when scanning some older photos into Immich as part of an external library.
The Import library workaround does not work if you never upload but use external libraries.

Cheers.

**Comment by berezki** [comment]
+1 for this one.
I download image from the internet and then I have to scroll through the whole list to find it (to put it in the album).

**Reply by Tech6767** [reply]
You can actually "reupload" an image directly into an album. Immich detects that the image is a duplicate and simply adds it to the album without having a duplicate file. Alternatively, you can do this the first time you upload manually. If you are talking about auto-uploads though, yeah, a "recents" feature is appreciated by me as well.

**Comment by sbopt** [comment]
+1 and also with support for adding assets to external libraries. 
Configurable time window with a sensible default of something like 30 days

Of course one implementation would be to just add a "added in the last X days" option to the search panel.

**Comment by T55189** [comment]
Adding my support to this feature request. Strongly desired. Many of my photos need to have their dates edited post-upload. This if very cumbersome without a 'recently added' search function.

**Comment by Talhamehar007** [comment]


**Reply by bo0tzz** [reply]
@Talhamehar007 please don't spam stuff like "+1" on threads, use the upvote button on the post instead.

**Comment by solumath** [comment]
Looking for this feature to be implemented.

When uploading/syncing assets, sometimes it happens that the asset is a duplicate but not detected by the algorithm. Looking it up can be painful, as I might not remember the date it was. I would appreciate a log with the last X uploaded assets so I can fast edit them if they were added by mistake, or I want to fast jump to them.

This happened to me mostly when trying Sync remote deletions [Experimental]. When if the app was not running on the phone, the deletions that happened using the browser would be undone upon opening of the app on the phone. As they would seem like they weren't backed up.

I know this is experimental, and I'm just looking into how to make it manageable and this feature would be perfect for it.

**Comment by Denyerec** [comment]
Ended up here as I was trying to see if I was being stupid, and couldn't find any simple way to just "Sort by import date" as I like the other users needed to verify if I'd tagged the whole batch of recent images.

Please please please "Sort Chronologically / Sort by Date Uploaded" on the main timeline!

**Comment by drix-1** [comment]
Yes, just started using Immich, and as I am adding a lot of pictures from different sources, times and so on, it would be great to sort by recent!

**Comment by brucethomax** [comment]
I mean, I would honestly settle for a link in the upload overlay. As of now, the successful uploads just vanish. https://github.com/immich-app/immich/discussions/26669

**Comment by S2ciOnur** [comment]
Opend 3 years ago and still no implementation?

**Reply by brucethomax** [reply]
What do you think about my comment above? Would that satisfy your needs or do you want a completely different view? Can you describe what you are looking for? Thanks! 

Take a look at [8716](https://github.com/immich-app/immich/discussions/26669)

**Reply by Denyerec** [reply]
I tend to use external libraries rather than direct uploads, so for me a "Recently added" view or "Sort by date added" option in the main view would be more useful than any upload overlay which I likely will never see :)

**Reply by AFumi39** [reply]
Also all the uploads form non-web sources (mobile, cli, immich-go, etc.) wouldn't be covered by any "upload overlay", so I think that this would be a useful "nice-to-have" feature, but cover only one specific scenario

**Reply by brucethomax** [reply]
Alright, what do you think of this? Adding a "Recents" to the Sidebar. This can be turned on an off via preferences to avoid clutter. Another option would be to add a options dropdown to Photos main area, though i couldn't find a clean place to put the dropdown.

The first two pictures are from a library where the rest were uploaded directly.

<img width="957" height="961" alt="Screenshot 2026-03-06 at 10 21 08" src="https://github.com/user-attachments/assets/cfff6cd9-c8cc-4e92-a0cf-037632e4257b" />

**Reply by s-martin** [reply]
Your suggestion would be a solution for me.

I also could think of a feature where the standard timeline can be sorted by a "added to immich" timestamp

**Reply by Sulfiore** [reply]
The thing in the sidebar is nice, I think the Apple way of doing it is better than hiding a button in the preferences. "Recently Added" Is on almost on the top on Apple Photos. 
And then put all the less used things in a library section. Like Trash, Archive, Locked Folder, those are barely used anyway. Of course the best thing would be to be able to personalize the sidebar to enable disable tabs and rearrange them but that's a bit more work.

**Reply by brucethomax** [reply]
> Your suggestion would be a solution for me.
> 
> I also could think of a feature where the standard timeline can be sorted by a "added to immich" timestamp

Yes, that is fine two. Can you recommend a place for it, since I couldn't find a place that would clutter the clean interface.

**Reply by roschlau** [reply]
Yeah, that added button in the sidebar is exactly how I would imagine this working. I'd call it "Recently Added" though, just to avoid any confusion regarding whether it's about being recently *added* or recently *taken*.

**Reply by brucethomax** [reply]
> The thing in the sidebar is nice, I think the Apple way of doing it is better than hiding a button in the preferences. "Recently Added" Is on almost on the top on Apple Photos. And then put all the less used things in a library section. Like Trash, Archive, Locked Folder, those are barely used anyway. Of course the best thing would be to be able to personalize the sidebar to enable disable tabs and rearrange them but that's a bit more work.

It's hard for me to follow this in text since I don't have Apple and I am not sure if you are referring to the iOS App (we are talking about web here). Can you provide screenshots? Be as specific as you can. The screenshot I provided shows it in the library section (though it might make more sense to move it below Albums).

**Reply by Sulfiore** [reply]
They actually have two sections for it, i've never realized. 

That's the most usefull one: the Imports sections, it shows what you imported when.
<img width="1222" height="693" alt="imports" src="https://github.com/user-attachments/assets/fa623e15-6f3b-46ac-afdd-ef537baffa97" />

And then there is the recently saved section, it shows what you've added from most old to most recent for a certain amount of time.
<img width="908" height="313" alt="recently_saved" src="https://github.com/user-attachments/assets/34e78d6d-ae95-4bae-b10d-5c98ddeed33e" />

But what I thought about the good way of doing it is showing it in the main menu on the left not hiding it somewhere. But that's subjective.

**Comment by bo0tzz** [comment]
I'm going to lock this thread, as the feature is very clear and has been exhaustively discussed, and any further comments are now just to express discontent about it not being in yet.

**Reply by brucethomax** [reply]
Yeah, I get that. I have left a new comment proposing what I have working locally. Let me know if can add anything here to be more helpful. Hit me up in #contributing if you want to go into details. Thanks!

**Comment by brucethomax** [comment]
**Proposed Design** 🚀
A new view, adding a "Recently Added" button to the sidebar. I've opted for this since it's a fairly standard feature in catalog tools. If you have better ideas, please feel free chime in with specifics.

<img width="1009" height="915" alt="Screenshot 2026-03-06 at 11 38 42" src="https://github.com/user-attachments/assets/9637a236-9274-40ad-b031-277f9a53613d" />


**Other Considerations Made**
_Adding sort by to Photos_ 😕
I tried adding a "Sort by date uploaded" to the Photos view, though I couldn't find a clean place to put it. If you think there is, please provide a screen shot or sketch.

_Adding sort by and date field options to Search_ 👀
I haven't tried adding a "Date field option" to the search, though this should be possible. It looks like the Search API already has input for this (though I haven't tested it)

**Propose Implementation**

- Add a new timeline view sorted by created
- Add a new route /recently-added
- Add a new field to the client/dto AssetDateField (open to naming suggestions)
```
export enum AssetDateField {
  /** Date taken / captured (localDateTime) */
  Taken = 'taken',
  /** Date uploaded to Immich (createdAt) */
  Uploaded = 'uploaded',
}
```
- Add an index to the `asset` table for createdAt, `asset_createdAt_month_idx`

**Optional - add field to asset details** 😄
I added this for verification, though it might be a interest addition. 
<img width="351" height="391" alt="Screenshot 2026-03-06 at 11 48 53" src="https://github.com/user-attachments/assets/7d17ed6e-fcaf-436a-83af-2c0c4e96a6c5" />


**Questions**
Show this be configurable for use to flip on? Should the be default on or off?

Let me know if I forgot something or should take into consideration anything else. I'll update this comment as I work through the comments.

**Reply by multi-suggester** [reply]
**Possible things/suggestions i can think of are =>**

- adding the recently addedd symbol to the right top beside the import button would be nice and convenient , but idk what the team would think about it 

- i think recently added button should be default if its added on right top and not default other wise

- adding a "go to" indicator for import boxes would be nice (by pressing that you would see the last batch or recently uploaded)

-date field can look like
photo date start - photo date end
import date start - import date end 
so that it wouldnt confuse in terms of name


- adding a small temprorary (glowing?) surrounding box or symbol that indicates its recently added to recently added photos can be a nice addition but idk what team would think either




**Could/would probably require a different PR =>**


- not showing the year bar on right side if all belongs to the same year(seems weird/unnecesary?)(or add months/weeks instead? (could probably require a seperate pr/rework?))

- it would probably require a different pr as well but itd be nice if we could see photos in terms of batches/imports ? (i mean google photos to immich  or adding 1000 photo to immich via web sort of batches ) 
(would be nice to have filter option so that we could seperate it like from cli , api , app , web sort of thing so that it wouldnt be cluttered(as we get photos from app sync all the time it would clutter the interface )) 
(it could also require google photos like batch minimization buttonyou can find this functionality in persons(just realized it doest exists in main timelinewhile searching for it) )
<img width="1872" height="67" alt="image" src="https://github.com/user-attachments/assets/f3963559-5808-40c6-adb1-0c4d072a43a1" />


- also i just realized it but i think we dont have sort function at all?? (would require a different pr as well prob since it needs to be integrated into many seperate places)

- also its a bit seperate but related ,  itd be nice if we had text scalability support for the side column at it keeps getting longer and i have to manually slide it every time (would probably require a different pr)

sorry for my poor articulation capability :sweat_smile: , i can get into more details or revarticulate it via chatgpt if you need

**Reply by S2ciOnur** [reply]
An symbol like a Library which call "recently uploaded" would be the perfect solution for me. It is about what i uploaded via drag and drop. Just to have an Overview. Currently it will be sorted into the timeline and i can not review my manual uploads. Thats the point.

**Reply by brucethomax** [reply]
I have posted a PR if ya'll want to pull it down and check it out. Would love some feedback! Thanks! ✌️

**Reply by s-martin** [reply]
#26800

**Reply by jaapjan14** [reply]
> I have posted a PR if ya'll want to pull it down and check it out. Would love some feedback! Thanks! ✌️

Hey @brucethomax, tried your PR on my instance (v2.6.0, pgvecto-rs pg15-v0.2.0, Docker build on QNAP). The timeline loads and I can see the upload-date view, but clicking any photo thumbnail opens a blank viewer and the page becomes unresponsive (requires full refresh).

Browser console shows:

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'year')
    at xUpjYVzi.js:2:52945
    at Wt (ef4g5249.js:1:22244)
    at le (ef4g5249.js:1:23314)
    at lt (ef4g5249.js:1:8636)
    at Z.process (ef4g5249.js:1:6253)

Looks like something is trying to read .year off a null/undefined date object when opening the asset viewer. Wanted to flag it — excited about this feature!

**Update:** I tracked down the bug. The `TimelineAsset` type was renamed from `localDateTime` → `adjustedDateTime` and `fileCreatedAt` → `originalDateTime` in `types.ts` and `month-group.svelte.ts`, but references in 8 other files weren't updated:

- `day-group.svelte.ts`
- `timeline-manager.svelte.ts`
- `month-group.svelte.ts`
- `search-support.svelte.ts`
- `ChangeDateAction.svelte`
- `thumbnail-util.ts`
- `Timeline.svelte`
- `timeline-util.ts` (`toTimelineAsset` function)

Patch that fixes all of them: https://gist.github.com/jaapjan14/e8aa6dd1b6e103c07903b6fd98d920e4

With the patch applied, the recently-added view works and clicking individual photos opens correctly.

**Comment by photoserver-pro** [comment]
Great to see this moving forward with a concrete proposal and a PR – this is exactly the kind of feature that makes a real difference in day-to-day usability.
That said, I'd like to add a broader perspective: this is not just about adding a "Recently Added" entry to the sidebar navigation. The underlying issue is more fundamental.
Sorting should be a universal capability available in every list/grid view in Immich – regardless of whether you are browsing Tags, People, Places, Albums, Folders, Search results, or a dedicated "Recently Added" view. Right now, sorting options are only available in the Albums view, which creates a deeply inconsistent experience across the entire application.
Typical sort dimensions that users consistently request across the community:

Capture date (ascending / descending) – the current default, but should be explicitly toggleable
Upload / added date – exactly what this discussion is about, and critically important for multi-source archives
Star rating – to surface the best photos first
File name (A→Z / Z→A) – essential for scanned archives without reliable EXIF data
File size – useful for storage management and finding large videos
Last modified date – important for curation workflows

This is discussed in detail in the community here:

[#1689](https://github.com/immich-app/immich/discussions/1689) – Sorting for albums (open since 2022)
[#10400](https://github.com/immich-app/immich/discussions/10400) – General sorting options for views

The core problem is always the same: once a result set grows beyond a few hundred assets – whether it's a tag with 6,000 photos, a person with 15,000 photos, or a recently imported archive – a flat, unsortable list becomes practically unusable. Sorting is not a comfort feature. It is a baseline navigation requirement.
The "Recently Added" view and the createdAt sort proposed here are a great first step. Ideally, this would be the beginning of a consistent sorting framework that rolls out across all views.
Strong +1 on the PR and the proposed direction. cc [@timonrieger](https://github.com/timonrieger)

**Comment by dfiorini** [comment]
Hi, i wonder how in 2026 we miss this feature, is so much important!

**Comment by bo0tzz** [comment]
#28272
